File: MarsRoverProb/src/com/marsroverprob/executer/MarsRoverExecutor.java
package com.marsroverprob.executer;

import com.marsroverprob.model.Rover;
import com.marsroverprob.parser.AbstractParser;
import com.marsroverprob.parser.RoverCommandsParser;
import com.marsroverprob.parser.RoverPositionParser;

public class MarsRoverExecutor implements AbstractExecutor {

  @Override
  public Rover execute(String roverPos, String roverCmds, String maxPos) {
    AbstractParser roverPosParser = new RoverPositionParser();
    AbstractParser roverCmdsParser = new RoverCommandsParser();
    Rover rover = new Rover();
    rover = roverPosParser.parse(rover, roverPos, maxPos);
    rover = roverCmdsParser.parse(rover, roverCmds, maxPos);
    System.out.println(rover.toString());
    return rover;
  }

}

File: MarsRoverProblem_C#/Source/marsroverprob/executor/MarsRoverExecutor.cs
using System;
using marsroverprob.executor.intf;
using marsroverprob.model;
using marsroverprob.parser.intf;
using marsroverprob.parser.posImpl;
using marsroverprob.parser.commdImpl;

namespace marsroverprob.executor.impl
{
    class MarsRoverExecutor : IAbstractExecutor
    {
        public Rover execute(String roverPos, String roverCmds, String maxPos){
                IAbstractParser roverPosParser = new RoverPositionParser();
                IAbstractParser roverCmdsParser = new RoverCommandsParser();
                Rover rover = new Rover(0,0,Orientation.N);
                roverPosParser.parse(rover, roverPos, maxPos);
                roverCmdsParser.parse(rover, roverCmds, maxPos);
                Console.WriteLine(rover);
                return rover;
        }
    }
}